doc/cprover-manual/loop-invariant-synthesizer.md

```diff
@@ -0,0 +1,28 @@
+[CPROVER Manual TOC](../)
+
+## Loop Invariant Synthesis with goto-instrument
+
+This is a loop invariant synthesizer built in goto-instrument that 
+synthesize [invariant clause](contracts-invariants.md) for a 
+pre-existing GOTO binary, annotate the invariant clause into 
+the GOTO binary, and produce the annotated binary as output.
+
+Loop invariant synthesized by the synthesizer must be inductive 
+in the sense that 
+1. the instrumented loop invariant checks success, no matter other 
+assertions fail or success;
+2. the base-case is from the whole GOTO program with a main 
+function as the entry point.
+
+For different synthesis strategy, there are different prerequisites
+on the input GOTO binary, and different guarantees on the synthesized
+loop invariants, summarized in the
+following table. 
+
+Strategy         | Flag                                           | Prerequisites                                      | Guarantees
+-----------------|------------------------------------------------|----------------------------------------------------|----
+Enumerative      | --synthesize-loop-invariants                   | The input GOTO binary contains only pointer checks | All checks success in the annotated GOTO binary
+
+## Additional Resources
```

Comment:
Are there any papers or external material about how this synthesis work that you could reference here too?

Reply:
Thank you for the review! 
There is not a perfect fit here. I link a related paper, and will write something in the training material which we can link later.